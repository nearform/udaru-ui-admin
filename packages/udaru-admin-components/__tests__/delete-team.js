import React from 'react'
import renderer from 'react-test-renderer'
import DeleteTeam from 'delete-team'

it('should render with minimum props', () => {
  const component = renderer.create(
    <DeleteTeam
      udaruUrl="udaru-url"
      authorization="auth"
      id="1"
      name="team name"
    />
  )
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

it('should warn user if no onCancel prop passed into component', () => {
  const component = renderer.create(
    <DeleteTeam
      udaruUrl="udaru-url"
      authorization="auth"
      id="1"
      name="team name"
    />
  )
  const instance = component.root.instance
  const spy = jest.spyOn(global.console, 'log')

  expect(typeof instance.props.onCancel).toBe('function')

  instance.props.onCancel()

  expect(spy).toHaveBeenCalled()
  expect(global.console.log.mock.calls[0][0]).toBe(
    'WARNING: No onCancel function passed into the <DeleteTeam /> component.'
  )

  spy.mockReset()
  spy.mockRestore()
})

it('should catch error in render', () => {
  const logError = jest.fn()
  const error = new Error('oops')
  const info = 'some info about error'
  const component = renderer.create(
    <DeleteTeam
      udaruUrl="udaru-url"
      authorization="auth"
      id="1"
      name="team name"
      logError={logError}
    />
  )
  const instance = component.root.instance

  instance.componentDidCatch(error, info)

  expect(instance.state.hasError).toBeTruthy()
  expect(instance.state.errorMessage).toBe('An error has occured.')
  expect(logError).toHaveBeenCalled()
  expect(logError).toHaveBeenCalledWith(error, info)
})

it('should handle dismissing alert message', async () => {
  const component = renderer.create(
    <DeleteTeam
      udaruUrl="udaru-url"
      authorization="auth"
      id="1"
      name="team name"
    />
  )

  const instance = component.root.instance

  instance.onDismiss()

  const { success, hasError, errorMessage } = instance.state

  expect(success).toBeFalsy()
  expect(hasError).toBeFalsy()
  expect(errorMessage).toBe('')
})

it('should set error message when no id passed in as a prop', () => {
  const component = renderer.create(
    <DeleteTeam udaruUrl="udaru-url" authorization="auth" name="team name" />
  )
  const instance = component.root.instance

  expect(instance.state.errorMessage).toBe('No Team ID passed into component.')
})

it('should set error message when no name passed in as a prop', () => {
  const component = renderer.create(
    <DeleteTeam udaruUrl="udaru-url" authorization="auth" id="team id" />
  )
  const instance = component.root.instance

  expect(instance.state.errorMessage).toBe(
    'No Team name passed into component.'
  )
})

it('should error with no url is set', () => {
  const formValues = {
    name: 'value',
    description: 'value'
  }
  const component = renderer.create(
    <DeleteTeam authorization="my-authorization-id" org="my-org" />
  )
  const instance = component.root.instance

  instance.onTeamDelete(formValues)
})

it('should error with no authorization is set', () => {
  const formValues = {
    name: 'value',
    description: 'value'
  }
  const component = renderer.create(
    <DeleteTeam udaruUrl="my-url" org="my-org" />
  )
  const instance = component.root.instance

  instance.onTeamDelete(formValues)
})

it('should delete a team successfully', async () => {
  global.fetch = jest.fn().mockImplementation(
    () =>
      new Promise((resolve, reject) => {
        resolve({
          ok: true
        })
      })
  )

  const formValues = {
    name: 'value',
    description: 'value'
  }
  const component = renderer.create(
    <DeleteTeam
      udaruUrl="http://my-udaru-url"
      authorization="my-authorization-id"
      org="my-special-org"
      id="123"
      name="team 123"
    />
  )
  const instance = component.root.instance
  await instance.onTeamDelete(formValues)

  const { success, hasError, errorMessage } = instance.state
  const [url, options] = global.fetch.mock.calls[0]

  const { method, headers } = options

  expect(url).toBe('http://my-udaru-url/authorization/teams/123')
  expect(method).toBe('DELETE')
  expect(headers).toEqual({
    authorization: 'my-authorization-id',
    org: 'my-special-org'
  })
  expect(success).toBeTruthy()
  expect(hasError).toBeFalsy()
  expect(errorMessage).toBe('')

  global.fetch.mockRestore()
})

it('should handle a failed delete', async () => {
  global.fetch = jest.fn().mockImplementation(
    () =>
      new Promise((resolve, reject) => {
        resolve({
          ok: false
        })
      })
  )

  const formValues = {
    name: 'value',
    description: 'value'
  }
  const component = renderer.create(
    <DeleteTeam
      udaruUrl="http://my-udaru-url"
      authorization="my-authorization-id"
      org="my-special-org"
      id="123"
      name="team 123"
    />
  )
  const instance = component.root.instance
  await instance.onTeamDelete(formValues)

  const { success, hasError, errorMessage } = instance.state

  expect(success).toBeFalsy()
  expect(hasError).toBeTruthy()
  expect(errorMessage).toBe('There was an error deleting that team.')

  global.fetch.mockRestore()
})
