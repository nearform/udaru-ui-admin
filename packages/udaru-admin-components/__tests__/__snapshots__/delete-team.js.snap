// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with minimum props 1`] = `
.emotion-0 {
  font-size: 48px;
  margin-top: .5em;
  margin-bottom: .5em;
}

.emotion-4 {
  font-size: 24px;
  margin-top: 1em;
  margin-bottom: 1em;
}

.emotion-10 {
  background: #eee;
  border-radius: 4px;
  border: 1px solid transparent;
  color: #333;
  display: inline-block;
  line-height: 1.5;
  padding: 6px 12px;
  text-align: center;
  -webkit-transition: all .2s cubic-bezier(0,.6,.3,1);
  transition: all .2s cubic-bezier(0,.6,.3,1);
  vertical-align: middle;
  white-space: nowrap;
  background-color: #d9534f;
  border-color: darken(.05,#d9534f);
  color: #fff;
}

.emotion-10:hover {
  background-color: #c9302c;
  border-color: #b52b27;
  color: #fff;
}

.emotion-12 {
  background: #eee;
  border-radius: 4px;
  border: 1px solid transparent;
  color: #333;
  display: inline-block;
  line-height: 1.5;
  padding: 6px 12px;
  text-align: center;
  -webkit-transition: all .2s cubic-bezier(0,.6,.3,1);
  transition: all .2s cubic-bezier(0,.6,.3,1);
  vertical-align: middle;
  white-space: nowrap;
  background-color: transparent;
  border-color: none;
}

.emotion-12:hover {
  color: #D8D8D8;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="sc-bdVaJa hnxqsY"
>
  <div
    className="emotion-2 emotion-3"
  >
    <h1
      className="emotion-0 emotion-1"
      fontSize={6}
    >
      Delete Team
    </h1>
    <hr />
  </div>
  <h3
    className="emotion-4 emotion-5"
    fontSize={4}
  >
    Are you sure you want to delete 
    team name
    ?
  </h3>
  <p
    className="emotion-8 emotion-9"
  >
    This action 
    <span
      className="emotion-6 emotion-7"
    >
      CANNOT
    </span>
     be undone.
  </p>
  <button
    className="emotion-10 emotion-11"
    onClick={[Function]}
    variant="danger"
  >
    DELETE
  </button>
  <button
    className="emotion-12 emotion-11"
    onClick={[Function]}
    variant="link"
  >
    CANCEL
  </button>
</div>
`;
